letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+
+In this tutorial you'll learn some of the basic concepts of Interledger and some of the standards that make ILP a protocol that can be applied to any payment network. 
+
+In the tutorial we will run a very basic online service which sells... letters, and write a client to pay for them.
+
+Every time you visit the site you will be given a random letter from A - Z. But, if you haven't paid the fee you'll be prompted to pay first and will only be able to collect your letter when you have done so.
+
+In order to make the payment required to get our letter we will also implement a simple client to make the payment.
+
+> This tutorial uses the XRP Testnet as the underlying payment network but any payment network can be used if there is a working ledger plugin for it. Check out the existing plugins that are in development on [GitHub](https://github.com/search?utf8=%E2%9C%93&q=ilp-plugin-).
+
+## What you need before you start:
+
+* a computer with an internet connection
+* [NodeJS](https://nodejs.org/en/download/current/) version 7 or higher installed
+* basic knowledge of the command line terminal
+* basic knowledge of JavaScript
+* (optional) have git installed
+
+## What you'll learn:
+
+* about conditional payments and hashtimelock agreements
+* about ILP Addresses
+* about the Ledger Plugin Interface (LPI) and how to configure and use plugins
+* how to build a service that accepts Interledger payments 
+* how to build a basic payment client for sending Interledger payments
+* (bonus) how to build a client proxy that automates paying for requests as required
+
+> **NOTE:** There will be times, while doing this tutorial, when following the instructions results in an error. **That may be intentional.** Read on to understand why the error exists and how to fix it.
+
+## Background
+
+To illustrate the basic ILP concepts we're going to build a simple online service that sells letters of the alphabet and accepts payments via ILP. *We haven't patented the idea so please be generous if you deploy this and become an instant millionaire.*
+
+To do this we need a **server**, run by the shop, and also a **client**, run by the customer. 
+
+### Server
+
+The server will perform two functions:
+1. It will host an HTTP service where clients can GET a new random letter with each request.
+1. It will listen for incoming ILP payments related to pending letter purchases, fulfill these to accept the payment, and provide the payer with a token they can use to claim their letter.
+
+Once we have completed the tutorial the server will perform the following steps:
+
+1. Connect to an account and monitor it for incoming payments
+1. Start a web server to accept letter requests
+1. Process a request, store the result, and provide the customer with details of how to pay for the request and get the result
+1. Process an incoming payment related to a previous request, and provide payer with details of how to perform request again using proof-of-payment token
```

Comment:
the server doesn't provide the payer with details of how to perform [the] request again, it just provides the proof-of-payment token (i.e. the fulfillment). The client then composes the instructions of how to get the letter using that fulfillment.